extension/src/lspClient/languageClient.ts

```diff
@@ -35,31 +28,24 @@ export class LanguageClientWrapper extends Disposable {
       )
     )
 
-    // Start the client. This will also launch the server
     this.start()
   }
 
   async start() {
-    await this.client.start()
-
-    const files = await findFiles('**/*.{yaml,json,py,toml}', '.??*')
-
-    const textDocuments = files.map(filePath => {
-      const uri = Uri.file(filePath).toString()
-      const languageId = filePath.endsWith('yaml') ? 'yaml' : 'json'
-      const text = readFileSync(filePath, 'utf8')
```

Comment:
The watcher would not have caused this issue. We already have at least one watcher which watches the entire workspace (and more).

Reply:
The more I think about this the more I feel like the approach was backwards. We should be parsing each `dvc.yaml` file when it is opened and checking to see which symbols are files. From there we read those files so that we can jump into them if needed.

Reply:
Relevant: https://code.visualstudio.com/updates/v1_72#_improved-startup-performance

See "NO MORE SYNC FS CALLS"